code/modules/clothing/masks/hailer.dm

```diff
@@ -1,10 +1,56 @@
+// **** Dispatch ****
+
+GLOBAL_LIST_EMPTY(sechailers)
+
+/datum/action/item_action/dispatch
+	name = "Signal dispatch"
+	desc = "Opens up a quick select wheel for reporting crimes, including your current location, to your fellow security officers."
+	button_icon_state = "dispatch"
+	icon_icon = 'icons/mob/actions/hailer_actions.dmi'
+
+/obj/item/clothing/mask/gas/sechailer
+	var/obj/item/radio/radio
+	var/radio_key = /obj/item/encryptionkey/headset_sec
+	var/radio_channel = "Security"
+	var/dispatch_cooldown = 20 SECONDS
+	var/last_dispatch = 0
+
+/obj/item/clothing/mask/gas/sechailer/Destroy()
+	QDEL_NULL(radio)
+	QDEL_NULL(radio_key)
+	GLOB.sechailers -= src
+	. = ..()
+
+/obj/item/clothing/mask/gas/sechailer/Initialize(mapload)
+	. = ..()
+	GLOB.sechailers += src
+	radio = new(src)
+	radio.keyslot = new radio_key
+	radio.listening = FALSE
+	radio.recalculateChannels()
+
+/obj/item/clothing/mask/gas/sechailer/proc/dispatch(mob/user)
+	if(world.time < last_dispatch + dispatch_cooldown)
+		to_chat(user, "<span class='notice'>Dispatch radio broadcasting systems are recharging.</span>")
+		return FALSE
+	var/list/options = list()
+	for(var/option in list("401 (murder)", "101 (resisting arrest)", "210 (breaking and entering)", "206 (riot)", "302 (assault on an officer)")) //Hardcoded for each icon, not all crimes need emergency callout for more officers
+		options[option] = image(icon = 'icons/effects/aiming.dmi', icon_state = option)
```

Comment:
I like this, but I agree with the comment above. The thing is going to be used for requesting backup, so the crime icon isn't all that important. The location should be the thing that is in the option wheel (so you have to choose a general location), or it should send out a backup request without the area provided to slow down the super fast response times (forcing people to check suit sensors via warden or something instead).

I like the idea of a more general location, so that if someone does you know somewhere near their department, but not quite their exact location which gives you an area to look and means there will be at least some delay before security comes swarming to your location.

Being that way also helps aleviate the sechailer magically always knowing where it is, even while off-network too since it relies on the player to know where they are.

Reply:
I am not planning to make any sort of pager/radio. Someone can make an addon of such on their own, but I have no plan of that. This is simply a port from NSV that I will update with some minor changes. Else someone else can take it.

Reply:
As for the idea of making it a general area instead of a specific area would defeat the purpose and would make it bothersome to use. It would also cause issues because on top of the few departments, you would need to figure out how to point out maints and other things which only some maps have making it obsolete real fast.

Reply:
You could make the location options fore, aft, port, and starboard. That'll cover you even outside of departments

Reply:
> You could make the location options fore, aft, port, and starboard. That'll cover you even outside of departments

code it yourself then, got no clue how I would do that.